File: Dynamic_Array/staticArray.py
"""
#   Author: Bryce Calhoun
#   Description: A basic Static Array class definition to use for a dynamic array implementation.
"""

class Static_Array_Exception(Exception):


    pass


class Static_Array:
    def __init__(self, capacity=10):
        """
        Initializing the size of the array and creating a list with the
        specified size
        """
        self.capacity = capacity
        self._data = [None] * capacity

        if capacity < 0:
            raise Static_Array_Exception("\nCapacity Must Be Positive !\n")


    def get_at_index(self, index) -> object:
        if index < 0 or index >= self.capacity:
            raise Static_Array_Exception("INVALID INDEX")

        value = self._data[index]
        return value

    def set_at_index(self, index, value):
        if index < 0 or index > self.capacity:
            raise Static_Array_Exception("INVALID INDEX")
        self._data[index] = value
        return

File: Static_Array/staticArray.cpp
/*
** Author: Bryce Calhoun
** Description: Static array object with basic get index and set index operations.
*/

#include <iostream>
using namespace std;


class StaticArray{
    /*
    Description:
    */
    private:
        int capacity;
        void** array;

    public:
        StaticArray(){
            /*
            Default Constructor
            */
            this->capacity = 10;
            this->array = new void*[this->capacity];
            return;
        }
        StaticArray(int cap){
            /*
            Non-default constructor
            */
            this->capacity = cap;
            this->array = new void*[this->capacity];
            return;
        }

        // accessor for capacity
        int get_capacity(){
            return this->capacity;
        }

        void set_at_index(int index, void* value){
            /*
            Description: Function to take an integer value and void pointer 
                        to set the void pointer in an index of the array
            */
            if(index < 0 || index >= this->capacity){
                return;
            }
            this->array[index] = value;
            return;
        }

        void* get_at_index(int index){
            /*
            Description: Function to take an integer index and return a void pointer 
                        from the array
            */
            if(index < 0 || index >= this->capacity){
                void* value = NULL;
                return value;
            }
            void* value = this->array[index];
            return value;      
        }
};




